src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Tests/ContextClasses.cs

```diff
@@ -126,6 +126,18 @@ public partial class ContextGenericContainer<T>
     {
         [JsonSerializable(typeof(JsonMessage))]
         public partial class NestedInGenericContainerContext : JsonSerializerContext { }
+
+        [JsonSerializable(typeof(JsonMessage))]
+        public partial class NestedGenericInGenericContainerContext<T1> : JsonSerializerContext { }
```

Comment:
Added test coverage is always nice to have, but I fail to use a real use case where `JsonSerializerContext` is generic. `JsonSerializableAttribute` types can't really bind to `T1` can they?

Reply:
> Added test coverage is always nice to have, but I fail to use a real use case where JsonSerializerContext is generic.

This change is related to the suggestion https://github.com/dotnet/runtime/pull/71619#discussion_r916640172.
To be honest, I don't know the use case. Maybe the type parameter will be useful in some overrides.

> JsonSerializableAttribute types can't really bind to T1 can they?

They can't, as far as I can see. Trying to use `T` in JsonSerializableAttribute breaks compilation:
`[JsonSerializable(typeof(List<T>))]`
> error CS0416: 'System.Collections.Generic.List': an attribute argument cannot use type parameters